app/src/main/kotlin/at/bitfire/davdroid/sync/SyncException.kt

```diff
@@ -11,41 +11,69 @@ import okhttp3.HttpUrl
  * Exception that wraps another notification together with potential information about
  * a local and/or remote resource that is related to the exception.
  */
-class SyncException(cause: Throwable): Exception(cause) {
+class SyncException(cause: Throwable) : Exception(cause) {
 
     companion object {
 
         // provide lambda wrappers for setting the local/remote resource
 
-        fun<T> wrapWithLocalResource(localResource: LocalResource<*>?, body: () -> T): T {
+        fun <T> wrapWithLocalResource(localResource: LocalResource<*>?, body: () -> T): T {
```

Comment:
Why not call the suspendable version using `runBlocking` here? As far as I can see they are the same no?

```
fun <T> wrapWithLocalResource(localResource: LocalResource<*>?, body: () -> T): T =
    runBlocking {
        wrapWithLocalResourceSuspending(localResource, body)
    }
```

Reply:
+1 I thought it's not possible because the `body` is suspending in one case. But it's of course possible to pass a non-suspending lambda as a suspending one.